### ISSUE TITLE
[BUG] Mid_registrar seems have some little issue

### ISSUE THREAD

**Issue by firestation** [OpenSIPS/opensips#2716]
**Opensips 3.2.2 and 3.2.4 latest**

**Describe the bug**
Mid_registrar shows errors in logs when i set mode = 2 in case of using registrar module too.  
...opensips[23693]: ERROR:mid_registrar:unregister_record: 'from' key not found, skipping De-REGISTER  
...opensips[23693]: ERROR:mid_registrar:mid_reg_aor_event: failed to unregister contact  

**To Reproduce**

I have found this issue with vanilla installation on centos 7 and debian 10. Opensips maybe 3.2.2 or 3.2.4
Example:
1. install vanilla version of 3.2
2. loadmodule "mid_registrar.so"
3. i am doing mid_registrar_save for msg with $tU like 10000 and for others i do "save()". 

**Expected behavior**

For all saved messaged error will rise. 

I suppose problem is in place where opensips saves contact\records. In case of using save function we don't save them in correct place. but mid_registrar rising callback function with any deleted\expired contacts. 
Thanks.

**Comment by liviuchircu** [comment]
Hello @firestation and thanks for the nice report!  I have to admit, I did little to no testing with `registrar.so` combined with `mid_registrar.so`, as each of them was tested individually.  When coupled together, although things _should_ work (they share the same user location API, after all), you may occasionally run into inconveniences such as the unnecessary ERROR logs you found.

I will try to push a fix for this specific scenario, but I don't promise I will test this setup much further than this.  After all, it has a conceptually bad design (you cannot have both a fighter airplane and a speed boat at the same time, you have to choose one and, no, a seaplane doesn't count :-) ).

**Comment by liviuchircu** [comment]
Hi, @firestation!  After a bit more thought, we shouldn't mask these mid-registrar ERROR logs, as they correctly point to a problem (error state) in the design: the `mid_registrar` is accessing contacts which it did not produce -- why not display an error in the logs to signify this?

So, to make things work, please use a separate `location` table for each registrar, if you plan on using the together on the same box.  For example, you could name them: `location` and `location_mid_reg`.

**Comment by firestation** [comment]
Many thanks for your check, 
I forgot to wrote about using 2 different tables for save and for mid_reg_save() but i am using 2 different tables, sure. 
my concept is simple, i am using ordinary registrar for one users, and mid_registrar for another users. 
Errors rising for all UN-REGISTER requests in "location" table wich was used only for save() function. 

example of code

```
        if (is_method("REGISTER")) {
                # store the registration and generate a SIP reply

            if ($tU =~ "^10{4}$") {
                xlog("L_INFO", "REGISTER $rU to main registrar...\n");
                mid_registrar_save("wss_location","c3f");
                switch ($retcode) {
                case 1:
                        xlog("L_INFO", "forwarding REGISTER to main registrar...\n");
                        $ru = "sip:157.245.47.36:5060";
                        if (!t_relay()) {
                            send_reply(500, "Server Internal Error 1");
                        }

                        break;
                case 2:
                        xlog("L_INFO", "REGISTER has been absorbed!\n");
                        break;
                default:
                    xlog("L_ERR", "mid-registrar error!\n");
                    send_reply(500, "Server Internal Error 2");
            }
            } else {
                xlog("L_INFO", "REGISTER myself $tU");
                        if (!save("location"))
                                xlog("$rc failed to register AoR $tu\n");
            }
                exit;
        }
```

**Comment by liviuchircu** [comment]
You are right, @firestation, as there is no concept of "own table(s)" in `mid_registrar`.  So the fix will have to include this support in one way or another (maybe a new modparam: "list of own tables"?!).

**Comment by firestation** [comment]
Sorry but i am not quite understood, Do you asking me about details or you just write your thoughts about how it may\should be fixed?

**Comment by liviuchircu** [comment]
No, it was just notes/thoughts.  This is, indeed an issue and I've re-opened it accordingly :-)

**Comment by liviuchircu** [comment]
@firestation I've just pushed two commits which should fix this issue on `master` branch: 30cdbd674 + c5d884e6.  Can you cherry-pick them into your sources and test them out?  For me, they solved the problem but having another confirmation before backporting them to the stable branches would be great to have.

Many thanks!
